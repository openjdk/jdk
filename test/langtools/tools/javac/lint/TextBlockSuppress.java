/*
 * @test /nodynamiccopyright/
 * @bug 8224228
 * @summary Verify SuppressWarnings works for LintCategore.TEXT_BLOCKS
 * @compile/fail/ref=TextBlockSuppress.out -XDrawDiagnostics -Xlint:text-blocks -Werror TextBlockSuppress.java
 */

public class TextBlockSuppress {

    public static class Example1 {
        public void method() {
            String s = """
                trailing space here:\u0020
                """;        // SHOULD get a warning here
        }
    }

    @SuppressWarnings("text-blocks")
    public static class Example2 {
        public void method() {
            String s = """
                trailing space here:\u0020
                """;        // SHOULD NOT get a warning here
        }
    }

    public static class Example3 {
        @SuppressWarnings("text-blocks")
        public void method() {
            String s = """
                trailing space here:\u0020
                """;        // SHOULD NOT get a warning here
        }
    }

    public static class Example4 {
        {
            String s = """
                trailing space here:\u0020
                """;        // SHOULD get a warning here
        }
    }

    @SuppressWarnings("text-blocks")
    public static class Example5 {
        {
            String s = """
                trailing space here:\u0020
                """;        // SHOULD NOT get a warning here
        }
    }

    public static class Example6 {
        public void method() {
            @SuppressWarnings("text-blocks")
            String s = """
                trailing space here:\u0020
                """;        // SHOULD NOT get a warning here
        }
    }
}
