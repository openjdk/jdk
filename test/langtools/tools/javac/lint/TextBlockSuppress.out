TextBlockSuppress.java:12:24: compiler.warn.trailing.white.space.will.be.removed
TextBlockSuppress.java:38:24: compiler.warn.trailing.white.space.will.be.removed
- compiler.err.warnings.and.werror
1 error
2 warnings
